doc/sphinx/language/core/primitive.rst

```diff
@@ -150,7 +150,17 @@ extraction. Instead, it has to be provided by the user (if they want to compile
 or execute the extracted code). For instance, an implementation of this module
 can be taken from the kernel of Coq (see ``kernel/parray.ml``).
 
-Primitive arrays expose a functional interface, but they are internally
-implemented using a persistent data structure :cite:`ConchonFilliatre07wml`.
-Update and access to an element in the most recent copy of an array are
-constant time operations.
+Semantically, the set array operation behaves as if the array was first entirely
+copied, then the copy is modified and finally returned. Operationally however,
```

Comment:
This change is good overall, but a few notes:

1. The beginning of the paragraph talks about “persistent” arrays — I’d say “purely functional” there, or some term making it clear that the semantics doesn’t use mutable state, or maybe add this sentence early.
2. When describing the time/space complexity, I’d still use the term “persistent data structure” because it’s the appropriate technical term, it just needs to be explained; it‘s the name for data structures where “multiple versions exist at the same time”.
3. YMMV, but IMHO “decorator” here doesn’t add much. The rest is good.

Reply:
@Blaisorblade Different keywords speak to different people. You are comfortable with "persistent data structures" so you want to use that word. Other people use "decorators" a lot and that will speak to them. Can you write here the exact paragraph that you want ?

Reply:
I don't expect all readers to know what persistent data structures are; but since that is the correct technical term, it should appear and be explained. I can try to suggest something later, but @jfehrle's the one in charge.

Reply:
> but @jfehrle's the one in charge.

I think that overstates the role of reviewers a bit, especially for documentation.  Good reviews often involve some discussion.  If any of my comments doesn't make sense or you disagree with them, please bring that up.  And there are certainly many things in Coq that I don't understand completely.

Reply:
I added "persistent data structures".